downstream.go

```diff
@@ -935,6 +944,15 @@ func (dc *downstreamConn) handleMessageRegistered(msg *irc.Message) error {
 		dc.forEachUpstream(func(uc *upstreamConn) {
 			uc.SendMessage(msg)
 		})
+
+		if dc.nick != nick {
```

Comment:
Hmm. So when downstream sends a NICK command, this code sends back an ack, but just for this downstream connection. I think we should just wait for the server ack instead?

Reply:
I'd also like the downstream nick to change if i'm connected to no upstreams or if i'm connected to multiple upstreams in multi-server mode. Right now the code in upstream.go only updates nicks when in single-server mode.

If you'd like we could change the downstream nick only if `dc.upstream() != nil` so that we wait for the server ack in that specific case. Although I don't think that's necessary. What I had in mind for the code in upstream.go was that when in single-server mode, if an OP force-nicks you to something else (a few servers let you do that), then you get that information and your downstream nick is updated. It wasn't about waiting for the server ACK.

What do you think?

Reply:
We need to make sure not to double-send the nick change.